### PR TITLE
fix [RAC]: RadioGroup/Dialog Focus

### LINKED ISSUE #4746: React Aria Components - RadioGroup/Dialog Focus Issue
# 🐛 Bug Report 

When React Aria Components `RadioGroup` is placed inside `Dialog`, pressing Tab each time, the focus moves to the first `Radio`, then the second `Radio`, e.t.c, instead of the first `Radio` to the next focusable element.

When `RadioGroup` exists outside `Dialog`, pressing Tab each time, the focus moves to the first `Radio` then the next focusable element.

## 🤔 Expected Behavior

`RadioGroup` focus should work the same inside and outside `Dialog`.

## 😯 Current Behavior

I recorded a demo. Ignore the styles and content. I copy and pasted everything from the Tailwind CSS examples. -> https://reactspectrum.blob.core.windows.net/reactspectrum/f239d0b1a96c3e6119135fe6bbf1994dc9984257/verdaccio/rac-tailwind/index.html

https://codesandbox.io/s/react-aria-compnoents-radio-group-dialog-focus-behavior-4f3xxc?file=/src/App.tsx

https://github.com/adobe/react-spectrum/assets/95544653/9bad4ca9-38c7-44d4-8086-2f9042ab559e

## 💁 Possible Solution

I'm guessing it has something to do with the `Dialog` focus trap. There's also a chance I've misunderstood something.

## 🔦 Context

I'm working on project, came across this issue, and was able to isolate/confirm it in CodeSandbox.

## 💻 Code Sample

https://codesandbox.io/s/react-aria-compnoents-radio-group-dialog-focus-behavior-4f3xxc?file=/src/App.tsx

## 🌍 Your Environment

I discovered it in Tauri, but have confirmed it's happening in Firefox and Chrome.

It's using `react-aria-components@1.0.0-alpha.5`.

## 🧢 Your Company/Team

Not with Adobe.

### PR DESCRIPTION
**Pull request by mariareedstrom**
Resolved issue with RadioGroup tab navigation within Dialog Component.

RadioGroup tab focus now complies with implementation of roving tabindex in accordance with [a11y radio guidelines](https://www.w3.org/TR/wai-aria-practices-1.1). 

Followed suggestion from @LFDanLu per https://github.com/adobe/react-spectrum/issues/4746#issuecomment-1629758418. Added Storybook examples from RadioGroup and RadioGroup in Dialog. Added tests to cover both use cases.

No change to documentation needed. 

Closes #4746 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### Visual test
1. Open Storybook
2. Select RadioGroup Example under React Aria Components
4. Tab to first radio, observe has focus
5. Tab again, observe focus no longer in RadioGroup
6. Repeat steps 2 - 5 for RadioGroup in Dialog Example

### Unit Test
Run newly added tests in `react-aria-components/test/RadioGroup.test.js`
- "should not navigate within the group using Tab"
- "should not navigate within the group using Tab in Dialog"

## 🧢 Your Project:

N/A

### CODE REVIEW

**Comment by LFDanLu** [comment]
GET_BUILD

**Review by reidbarber — DISMISSED** [review]
LGTM, and thanks for the stories + tests!

File: packages/react-aria-components/example/index.css [hunk 1]
```diff
@@ -206,3 +206,85 @@ html {
...
+      border-color: var(--invalid-color);
+    }
+
+    &[data-pressed]:before {
+      border-color: var(--invalid-color-pressed);
+    }
+  }
+
+  &[data-focus-visible]:before {
+    box-shadow: 0 0 0 2px var(--spectrum-alias-background-color-default), 0 0 0 4px var(--selected-color);
```
Comment (LFDanLu) [inline packages/react-aria-components/example/index.css:284]:
```suggestion
    box-shadow: 0 0 0 2px canvas, 0 0 0 4px var(--selected-color);
```

**Review by LFDanLu — APPROVED** [review]
LGTM, thank you for the thorough tests and stories! Pushed a small change to get the radio focus ring rendering in the dialog, hope you don't mind.

**Review by yihuiliao — APPROVED** [review]
lgtm, thank you for your contribution!

**Review by reidbarber — APPROVED** [review]

**Pull request merged**